shell/platform/fuchsia/flutter/accessibility_bridge.cc

```diff
@@ -23,6 +23,7 @@ AccessibilityBridge::AccessibilityBridge(
   fuchsia_semantics_manager_.set_error_handler([](zx_status_t status) {
     FML_LOG(ERROR) << "Flutter cannot connect to SemanticsManager with status: "
                    << zx_status_get_string(status) << ".";
+    FML_CHECK(status != ZX_ERR_PEER_CLOSED);
```

Comment:
This will crash the entire process (which may contain multiple flutter apps).

Instead, let's do this:
1.  Find the `session_listener_error_callback` in shell/platform/fuchsia/flutter/platform_view.cc
2.  Plumb it through to the AccessibilityBridge and call it from this method.
3.  Be prepared to fix lots of upstream tests that don't have a11y_mgr.cmx in their environment, and so will now crash. (you wouldve had to do this with the FML_CHECK)

Reply:
Happy to discuss this further offline. This is something that @neelsa and I went back and forth about for a while. I was initially raising the exact same concern.

This is basically equivalent to Scenic telling the flutter_runner that it's disconnecting it. Flutter_runner only maintains one connection to the a11y bridge for the process - it's not one connection per application. If the manager disconnects the runner, there is no good path to recovery, and for an a11y user any flutter application becomes unusable (much like for a sighted user if we can't render anything with Scenic).  This is a "this should never happen" bug, but crashing (and getting crash reports) seems like the right thing to do here.

Individual apps should not need a11y_mgr.cmx, it's in the cmx file for flutter_runner which establishes and maintains the connection. Again, this was done with the idea that it makes no sense for a Flutter application to opt-out of a11y, any more than it makes sense to opt-out of scenic rendering.

Reply:
Just to add - if an application is capable of causing a disconnect here, there is either a bug in the a11y_manager or flutter_runner - we should never be sending data to a11y so invalid that it disconnects us.